pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -165,3 +167,9 @@ func SetDefaults_NetworkPolicy(obj *extensionsv1beta1.NetworkPolicy) {
 		}
 	}
 }
+
+func SetDefaults_HTTPIngressPath(obj *extensionsv1beta1.HTTPIngressPath) {
+	if obj.PathType == nil {
+		obj.PathType = &defaultPathType
```

Comment:
Prefer a pointer to a local value. Returning a pointer to a package variable here means that any caller that assigns to that pointer will override the global default value. See https://play.golang.org/p/rGUjo2eB8dz